test/src/main/java/org/apache/accumulo/test/FateSummaryIT.java

```diff
@@ -174,93 +187,32 @@ public void testFatePrintAndSummaryCommandsWithInProgressTxns() throws Exception
-        // mid-print.
-        do {
-          // Execute the command when transactions are currently running and may complete mid-print
-          p = execAdminFateCommand(command);
-          // Previously, this check could fail due to a ZK NoNodeException
-          assertEquals(0, p.getProcess().waitFor());
-          result = p.readStdOut();
-          // A transaction should have completed mid-print and been ignored
-        } while (!result.contains("Tried to get info on a since completed transaction - ignoring")
-            && !noTransactions(result, command));
-
-        if (noTransactions(result, command)) {
-          // Fail since we printed until all transactions have completed and didn't see a
-          // transaction complete mid-print.
-          // This is highly unlikely to have occurred.
-          fail();
-        }
-        // Otherwise, we saw 'Tried to get info on a since completed transaction - ignoring', so
-        // test passes
-        client.tableOperations().delete(table);
-      }
-    }
-  }
-
-  private boolean noTransactions(String result, String command) {
-    if (command.equals("--print")) {
-      return result.contains(" 0 transactions");
-    } else { // --summary
-      result = result.substring(result.indexOf("{"), result.lastIndexOf("}") + 1);
-      FateSummaryReport report = FateSummaryReport.fromJson(result);
-      return report != null && report.getReportTime() != 0 && report.getStatusCounts().isEmpty()
-          && report.getStepCounts().isEmpty() && report.getCmdCounts().isEmpty()
-          && report.getStatusFilterNames().isEmpty() && report.getFateDetails().isEmpty();
-    }
-  }
-
-  private ProcessInfo execAdminFateCommand(String command) throws Exception {
-    if (command.equals("--print")) {
-      return getCluster().exec(Admin.class, "fate", command);
-    } else { // --summary
-      return getCluster().exec(Admin.class, "fate", command, "-j");
+    ServerContext sctx = getCluster().getServerContext();
+
+    // This error was occurring in AdminUtil.getTransactionStatus(). One of the methods that is
+    // called which may throw the NNE is top(), so we will mock this method to always throw a NNE
+    // and ensure it is handled/ignored within getTransactionStatus()
+    ZooStore<String> zs = EasyMock.createMockBuilder(ZooStore.class)
+        .withConstructor(String.class, ZooReaderWriter.class)
+        .withArgs(sctx.getZooKeeperRoot() + Constants.ZFATE, sctx.getZooReaderWriter())
+        .addMockedMethod("top").createMock();
+    expect(zs.top(anyLong())).andThrow(new RuntimeException(new KeeperException.NoNodeException()))
+        .atLeastOnce();
+    replay(zs);
+    zs.create();
+
+    Class<?> clazz = AdminUtil.class;
+    Object adminUtil = clazz.getDeclaredConstructor(boolean.class).newInstance(true);
+    Method getTransactionStatus = clazz.getDeclaredMethod("getTransactionStatus",
```

Comment:
Was wondering if would be easy to change this method to be public static to avoid reflection.  Tried it locally and it required very minimal changes.  If that could work it would be better than reflection because we would get compile error for the test when something changes.

```diff
diff --git a/core/src/main/java/org/apache/accumulo/core/fate/AdminUtil.java b/core/src/main/java/org/apache/accumulo/core/fate/AdminUtil.java
index b7afea1b84..3ea322ef78 100644
--- a/core/src/main/java/org/apache/accumulo/core/fate/AdminUtil.java
+++ b/core/src/main/java/org/apache/accumulo/core/fate/AdminUtil.java
@@ -333,7 +333,7 @@ public class AdminUtil<T> {
    * @param waitingLocks populated list of locks held by transaction - or an empty map if none.
    * @return current fate and lock status
    */
-  private FateStatus getTransactionStatus(ReadOnlyTStore<T> zs, Set<Long> filterTxid,
+  public static <T> FateStatus getTransactionStatus(ReadOnlyTStore<T> zs, Set<Long> filterTxid,
       EnumSet<TStatus> filterStatus, Map<Long,List<String>> heldLocks,
       Map<Long,List<String>> waitingLocks) {
 
@@ -399,11 +399,11 @@ public class AdminUtil<T> {
 
   }
 
-  private boolean includeByStatus(TStatus status, EnumSet<TStatus> filterStatus) {
+  private static boolean includeByStatus(TStatus status, EnumSet<TStatus> filterStatus) {
     return (filterStatus == null) || filterStatus.contains(status);
   }
 
-  private boolean includeByTxid(Long tid, Set<Long> filterTxid) {
+  private static boolean includeByTxid(Long tid, Set<Long> filterTxid) {
     return (filterTxid == null) || filterTxid.isEmpty() || filterTxid.contains(tid);
   }


```

Reply:
Yeah, I also think that would be better. I can change this. Was worried about making something public that shouldn't be

Reply:
The package the class is in does not fall under the public API so its ok to make it public.  Could add a `@VisibleForTesting` annotation.